mediator/med_enthalpy_mod.F90

```diff
@@ -0,0 +1,175 @@
+    else
+       call FB_GetFldPtr(is_local%wrap%FBExp(compatm), 'Faxx_evap' , evap, rc=rc)
+       if (ChkErr(rc,__LINE__,u_FILE_u)) return
+    endif
+    
+    if(FB_fldchk(is_local%wrap%FBExp(compocn), 'Foxx_hevap', rc)) then
+       call FB_GetFldPtr(is_local%wrap%FBExp(compocn), 'Foxx_hevap', hevap, rc=rc)
+       if (ChkErr(rc,__LINE__,u_FILE_u)) return
+    else
+       allocate(hevap(nmax))
+    endif
+    if(FB_fldchk(is_local%wrap%FBExp(compocn), 'Foxx_hcond', rc)) then
+       call FB_GetFldPtr(is_local%wrap%FBExp(compocn), 'Foxx_hcond', hcond, rc=rc)
+       if (ChkErr(rc,__LINE__,u_FILE_u)) return
+    else
+       allocate(hcond(nmax))
+    endif
+
+    if(FB_fldchk(is_local%wrap%FBExp(compocn), 'Faxa_snow', rc)) then
+       call FB_GetFldPtr(is_local%wrap%FBExp(compocn), 'Faxa_snow' , snow, rc=rc)
+       if (ChkErr(rc,__LINE__,u_FILE_u)) return
+    else
+       call FB_GetFldPtr(is_local%wrap%FBImp(compatm,compatm), 'Faxa_snowc' , snowc, rc=rc)
+       if (ChkErr(rc,__LINE__,u_FILE_u)) return
+       call FB_GetFldPtr(is_local%wrap%FBImp(compatm,compatm), 'Faxa_snowl' , snowl, rc=rc)
+       if (ChkErr(rc,__LINE__,u_FILE_u)) return
+       allocate(snow(nmax))
+       snow = snowc + snowl
+    endif
+
+    if(FB_fldchk(is_local%wrap%FBExp(compocn), 'Foxx_hsnow', rc)) then
+       call FB_GetFldPtr(is_local%wrap%FBExp(compocn), 'Foxx_hsnow', hsnow, rc=rc)
+       if (ChkErr(rc,__LINE__,u_FILE_u)) return
+    else
+       allocate(hsnow(nmax))
+    endif
+    
+    if(FB_fldchk(is_local%wrap%FBExp(compocn), 'Foxx_rofl', rc)) then
+       call FB_GetFldPtr(is_local%wrap%FBExp(compocn), 'Foxx_rofl' , rofl, rc=rc)
+       if (ChkErr(rc,__LINE__,u_FILE_u)) return
+    else
+       call FB_GetFldPtr(is_local%wrap%FBImp(comprof, comprof), 'Forr_rofl', rofl, rc=rc)
+       if (ChkErr(rc,__LINE__,u_FILE_u)) return
+    endif
+    if(FB_fldchk(is_local%wrap%FBExp(compocn), 'Foxx_hrofl', rc)) then
+       call FB_GetFldPtr(is_local%wrap%FBExp(compocn), 'Foxx_hrofl', hrofl, rc=rc)
+       if (ChkErr(rc,__LINE__,u_FILE_u)) return
+    else
+       allocate(hrofl(nmax))
+    endif
+
+    if(FB_fldchk(is_local%wrap%FBExp(compocn), 'Foxx_rofi', rc)) then
+       call FB_GetFldPtr(is_local%wrap%FBExp(compocn), 'Foxx_rofi' , rofi, rc=rc)
+       if (ChkErr(rc,__LINE__,u_FILE_u)) return
+    else
+       call FB_GetFldPtr(is_local%wrap%FBImp(comprof, comprof), 'Forr_rofi', rofi, rc=rc)
+       if (ChkErr(rc,__LINE__,u_FILE_u)) return
+    endif
+    if(FB_fldchk(is_local%wrap%FBExp(compocn), 'Foxx_hrofi', rc)) then
+       call FB_GetFldPtr(is_local%wrap%FBExp(compocn), 'Foxx_hrofi', hrofi, rc=rc)
+       if (ChkErr(rc,__LINE__,u_FILE_u)) return
+    else
+       allocate(hrofi(nmax))
+    endif
+    do n = 1,nmax
+       ! Need max to ensure that will not have an enthalpy contribution if the water is below 0C
+       hrain(n)  = max((tocn(n) - tkfrz), 0._r8) * rain(n)  * cpfw
+       hsnow(n)  = min((tocn(n) - tkfrz), 0._r8) * snow(n)  * cpice
```

Comment:
Please change
```
hrain(n)  = max((tocn(n) - tkfrz), 0._r8) * rain(n)  * cpfw
￼hsnow(n)  = min((tocn(n) - tkfrz), 0._r8) * snow(n)  * cpice
```
to 
```
hrain(n)  = max((tbot(n) - tkfrz), 0._r8) * rain(n)  * cpfw
￼hsnow(n)  = min((tbot(n) - tkfrz), 0._r8) * snow(n)  * cpice
```
where **tbot** is 

```
if  (FB_fldchk(is_local%wrap%FBExp(compocn), 'Sa_tbot'    , rc=rc)) then
         call FB_GetFldPtr(is_local%wrap%FBExp(compocn), 'Sa_tbot', tbot, rc=rc)
         if (ChkErr(rc,__LINE__,u_FILE_u)) return
endif
```

Reply:
done - Note that with these changes it is no longer possible to compare to b.e23_alpha16b.BLT1850.ne30_t232.040